src/azure-cli/azure/cli/command_modules/monitor/_params.py

```diff
@@ -339,12 +339,53 @@ def load_arguments(self, _):
+                   arg_type=get_enum_type(PublicNetworkAccessType))
 
     with self.argument_context('monitor log-analytics workspace pack') as c:
         c.argument('intelligence_pack_name', options_list=['--name', '-n'])
         c.argument('workspace_name', options_list='--workspace-name')
     # endregion
 
+    # region Log Analytics Workspace Linked Service
+    with self.argument_context('monitor log-analytics workspace linked-service') as c:
+        c.argument('linked_service_name', name_arg_type, help='Name of the linkedServices resource. Supported values: cluster, automation.')
+        c.argument('workspace_name', options_list='--workspace-name')
+        c.argument('resource_id', help='The resource id of the resource that will be linked to the workspace. This '
+                                       'should be used for linking resources which require read access.')
+        c.argument('write_access_resource_id', help='The resource id of the resource that will be linked to the '
+                                                    'workspace. This should be used for linking resources which '
+                                                    'require write access.')
+    # endregion
+
+    # region Log Analytics Cluster
+    with self.argument_context('monitor log-analytics cluster') as c:
+        c.argument('cluster_name', name_arg_type, help='The name of the Log Analytics cluster.')
+        c.argument('sku_name', help="The name of the SKU. Currently only support 'CapacityReservation'")
+        c.argument('sku_capacity', help='The capacity of the SKU. It must be in the range of 1000-2000 per day and must'
+                                        ' be in multiples of 100. If you want to increase the limit, please contact'
+                                        ' LAIngestionRate@microsoft.com. It can be decreased only after 31 days.')
+        c.argument('identity_type', help='The identity type. Supported values: SystemAssigned')
+
+    with self.argument_context('monitor log-analytics cluster update') as c:
+        c.argument('key_vault_uri', help='The Key Vault uri which holds the key associated with the Log Analytics cluster.')
+        c.argument('key_name', help='The name of the key associated with the Log Analytics cluster.')
+        c.argument('key_version', help='The version of the key associated with the Log Analytics cluster.')
+    # endregion
+
+    # region Log Analytics Linked Storage Account
+    with self.argument_context('monitor log-analytics workspace linked-storage') as c:
+        from azure.mgmt.loganalytics.models import DataSourceType
+        c.argument('data_source_type', help='Data source type for the linked storage account.',
+                   options_list=['--type'], arg_type=get_enum_type(DataSourceType))
+        c.argument('storage_account_ids', nargs='+', options_list=['--storage-accounts'],
```

Comment:
will it be required for create_or_update?

Reply:
required. I think it make senses. Creating a linked storage without storage account is quite strange.